pkg/asset/installconfig/gcp/client.go

```diff
@@ -222,6 +224,34 @@ func (c *Client) GetProjects(ctx context.Context) (map[string]string, error) {
 	return projects, nil
 }
 
+// GetProjectRegions gets the regions that are valid for the project. An error is returned when unsuccessful
+func (c *Client) GetProjectRegions(ctx context.Context, project string) ([]string, error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	gc, err := google.DefaultClient(ctx, compute.CloudPlatformScope)
+	if err != nil {
+		return nil, err
+	}
+
+	service, err := compute.New(gc)
+	if err != nil {
+		return nil, err
+	}
+
+	r, err := service.Regions.List(project).Context(ctx).Do()
+	if err != nil {
+		return nil, err
```

Comment:
Some of these:

```go
return nil, err
```
will go away if we change how the service is instantiated, but in general when errors are returned like this we should `wrap` them so we have an idea of where failures are occurring when errors happen.